php/commands/site.php

```diff
@@ -994,7 +996,7 @@ public function deploy($args, $assoc_args) {
       'annotation'     => $annotation,
     );
 
-    if ($clone_live_content) {
+    if ($sync_content) {
```

Comment:
It's already only valid in the test environment:

https://github.com/pantheon-systems/cli/blob/site-deploy%23463/php/commands/site.php#L978

I considered adding a warning if used on live environment but felt that was needlessly complain-y. Doesn't seem worth yelling at the user for doing something harmless.

I could be convinced it's helpful though. Especially since the flag no longer has the word 'live' in it.